test/util-test.js

```diff
@@ -434,4 +434,41 @@ describe('util.addFormatMeta()', () => {
       });
     });
   });
+  describe('util.cutAfterJSON()', () => {
+    it('works with simple JSON', () => {
+      assert.equal(util.cutAfterJSON('{"a": 1, "b": 1}'), '{"a": 1, "b": 1}');
+    });
+    it('Cut extra characters after JSON  end', () => {
+      assert.equal(util.cutAfterJSON('{"a": 1, "b": 1}abcd'), '{"a": 1, "b": 1}');
+    });
+    it('tolerant to string constants', () => {
+      assert.equal(util.cutAfterJSON('{"a": "}1", "b": 1}abcd'), '{"a": "}1", "b": 1}');
+    });
+    it('tolerant to string with escaped quoting', () => {
+      // eslint-disable-next-line no-useless-escape
+      assert.equal(util.cutAfterJSON('{"a": "\\\"}1", "b": 1}abcd'), '{"a": "\\\"}1", "b": 1}');
+    });
+    it('works with nested', () => {
+      assert.equal(
+        // eslint-disable-next-line no-useless-escape
+        util.cutAfterJSON('{"a": "\\\"1", "b": 1, "c": {"test": 1}}abcd'),
+        // eslint-disable-next-line no-useless-escape, comma-dangle
+        '{"a": "\\\"1", "b": 1, "c": {"test": 1}}'
+      );
+    });
+    it('works with utf', () => {
+      assert.equal(
+        // eslint-disable-next-line no-useless-escape
+        util.cutAfterJSON('{"a": "\\\"фыва", "b": 1, "c": {"test": 1}}abcd'),
+        // eslint-disable-next-line no-useless-escape, comma-dangle
+        '{"a": "\\\"фыва", "b": 1, "c": {"test": 1}}'
+      );
+    });
+    it('works with \\\\ in string', () => {
+      assert.equal(
+        util.cutAfterJSON('{"a": "\\\\фыва", "b": 1, "c": {"test": 1}}abcd'),
+        '{"a": "\\\\фыва", "b": 1, "c": {"test": 1}}',
+      );
+    });
```

Comment:
I won't do one with uneven brackets but rather with missing closing brackets, due to the new way the function handles cutting the JSON now. The function should never encounter more closing brackets than opening brackets because the function starts with a single opening bracket and will cut the JSON as soon as it finds an equal amount of closing brackets.

One with a semi-colon ending would be unnecessary because there is already a "Cut extra characters after JSON end" test and the function now doesn't differentiate between a semi-colon and any other character anymore.

Reply:
ok cool